presto-main/src/main/java/io/prestosql/operator/OrderByOperator.java

```diff
@@ -148,6 +147,8 @@ public OperatorFactory duplicate()
     private enum State
     {
         NEEDS_INPUT,
+        SECONDAY_INPUT,
```

Comment:
instead of secondary input & blocked..  just add a new state as Finishing.

```java
needInput() {
     return (primarySpill.isDone() || secondarySpill.isDone) && state == NEEDS_INPUT;
}
addInput(page) {
     if (primarySpill.isDone()) { 
            primaryIndex.addPage(page);
            if (secSpill.isDone()  && secIndex.hasData()) {
                triggerSecondarySpill();
            }
     } else {
          secIndex.addPage(page);
          if (secIndex == full)
              triggerSecondarySpill();
     }

    checkAnyBackgroundSpillFinish(); // call finishRevoke for secSpills in this
}

finish() {
     if (state == HAS_OUTPUT) {
          return;
     }
     if (primarySpill.isDone() && secSpill.isDone() && (state == NEEDS_INPUT || state == FINISHING)) {
            state = HAS_OUTPUT;
            ....
     }
     if (state == FINISHING) {
           checkAnyBackgroundSpillFinish();
           return;
     }
     state = FINISHING;
}
```